src/modules/formBuilder/components/FormBuilder.tsx

```diff
@@ -0,0 +1,224 @@
+            setUnblockCallback(undefined);
+          });
+        }}
+        onCancel={() => {
+          setIsBlocked(false);
+          setUnblockCallback(undefined);
+        }}
+        maxWidth='sm'
+        fullWidth
+      >
+        <Typography>
+          You have unsaved changes. Save before proceeding?
+        </Typography>
+      </ConfirmationDialog>
+      <Box sx={{ display: 'flex' }}>
+        <FormBuilderPalette
+          onItemClick={(itemType) => {
+            const newItem: FormItem = {
+              linkId: crypto.randomUUID(), // TODO(#6083) - this is a placeholder, but right now will only work in localhost
+              text: itemType.toString(),
+              type: itemType,
+              required: false,
+              warnIfEmpty: false,
+              hidden: false,
+              readOnly: false,
+              repeats: false,
+              prefill: false,
+              disabledDisplay: DisabledDisplay.Hidden,
+              enableBehavior: EnableBehavior.Any,
+            };
+
+            const newDefinition: FormDefinitionJson = {
+              ...workingDefinition,
+              item: [...workingDefinition.item, newItem],
+            };
+
+            setWorkingDefinition(newDefinition);
+          }}
+        />
+        {selectedItem && originalLinkId && (
+          <FormItemEditor
+            selectedItem={selectedItem}
+            originalLinkId={originalLinkId}
+            definition={formDefinition}
+            saveLoading={saveLoading}
+            errorState={errorState}
+            onSave={(item, originalLinkId) => {
+              const newDefinition = updateFormItem(
+                workingDefinition,
+                item,
+                originalLinkId
+              );
+
+              onSave(newDefinition);
+            }}
+            onDiscard={closeItemEditor}
+          />
+        )}
+        <Box
+          sx={
+            // Padding matches the padding usually applied in DashboardContentContainer.
+            // (It's moved in here because of the drawer)
+            {
+              flexGrow: 1,
+              pt: 2,
+              pb: 8,
+              px: { xs: 1, sm: 3, lg: 4 },
+            }
+          }
+        >
+          <FormBuilderHeader
+            formDefinition={formDefinition}
+            lastUpdatedDate={lastUpdatedDate}
+            onClickPreview={onClickPreview}
+          />
+          <Box sx={{ p: 4 }}>
+            <FormTree
+              definition={workingDefinition}
+              onEditClick={(item: FormItem) => {
+                function editItem() {
+                  setSelectedItem(item);
+                  setOriginalLinkId(item.linkId);
+                }
+
+                if (dirty) {
+                  setIsBlocked(true);
+                  setUnblockCallback(() => editItem);
+                } else {
+                  editItem();
+                }
+              }}
```

Comment:
Bringing together a couple of my previous comments, this could be streamlined into
```
if (dirty) {
   // form has unsaved changes, so prompt save before proceeding to edit
   setBlockedAction(() => setSelectedItem(item))  //<= no  need for second piece of state
} else {
   setSelectedItem(item); //<== no need for link id state
}
```